incubator/hnc/pkg/reconcilers/object.go

```diff
@@ -91,22 +95,53 @@ func (r *ObjectReconciler) SyncNamespace(ctx context.Context, log logr.Logger, n
 	return nil
 }
 
+// SyncCluster can be called manually by the ConfigReconciler when the mode of a type is changed.
+// It enqueues all the current objects in all namespaces.
+func (r *ObjectReconciler) SyncCluster(ctx context.Context, log logr.Logger) error {
+	log = log.WithValues("gvk", r.GVK)
+
+	keys := r.Forest.GetNamespaceNames()
+	for _, ns := range keys {
+		// Enqueue all the current objects in the namespace.
+		if err := r.enqueueLocalObjects(ctx, log, ns); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 // GetGVK provides GVK that is handled by this reconciler.
 func (r *ObjectReconciler) GetGVK() schema.GroupVersionKind {
 	return r.GVK
 }
 
+// GetMode provides mode of objects that are handled by the reconciler.
+func (r *ObjectReconciler) GetMode() api.SynchronizationMode {
+	return r.Mode
+}
+
+// SetMode provides mode of objects that are handled by the reconciler.
+func (r *ObjectReconciler) SetMode(mode api.SynchronizationMode) {
+	r.Mode = mode
```

Comment:
Shouldn't we validate the value here? Below, we do a direct comparison to `if r.Mode == api.Ignore`, which means that if I accidentally say `mode: "ingore"`, the objects will _still_ be propagated.

Reply:
Right now we validate the mode when we get it from the spec (see `getMode` in `hnc_config.go`). Then we call `SetMode` and trigger the `Reconcile`. The mode should already be validated when we do the `if r.Mode == api.Ignore` comparison in `Reconcile` of `object.go`.

Reply:
Yup, I agree that this works today. I'm just worried about accidentally adding a new codepath in the future that bypasses the check.